### PR TITLE
Add link prop to main menu to manage routing pages

### PR DESCRIPTION
**Pull request by dtrucs**
In order to make MainMenu component agnostic, we need to remove the `react-router-dom` dependencies from the project.
With this PR, the NavLink component (or whatever other link component) must be forwarded directly from the project using it to the MainMenu component.

### CODE REVIEW

File: src/components/HeaderLink.js [hunk 1]
```diff
@@ -1,34 +1,50 @@
...
+      linkProps: { hrefAttribute = 'href', ...linkProps },
+    } = link;
+
+    const attributesLink = {
+      ...attributes,
+      [hrefAttribute]: href,
+      ...linkProps,
+    };
+
+    return <Link {...attributesLink}>{children}</Link>;
```
Comment (mabhub) [inline src/components/HeaderLink.js:27]: I do not know if it is better or not. I was just giving it a try:

```suggestion
  if (link && !href.startsWith('http')) {
    const {
      component: Component,
      linkProps: { hrefAttribute = 'href', ...linkProps },
    } = link;

    return (
      <Component
        {...attributes}
        {...{ [hrefAttribute]: href }}
        {...linkProps}
      >
        {children}
      </Component>
    );
```

Comment (mabhub) [inline src/components/HeaderLink.js:27]: I do not know if it is better or not. I was just giving it a try:

```suggestion
    const {
      component: Component,
      linkProps: { hrefAttribute = 'href', ...linkProps },
    } = link;

    return (
      <Component
        {...attributes}
        {...{ [hrefAttribute]: href }}
        {...linkProps}
      >
        {children}
      </Component>
    );
```

File: src/components/HeaderLink.js [hunk 2]
```diff
@@ -1,34 +1,50 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import { NavLink } from 'react-router-dom';
 import classnames from 'classnames';
 
-export const HeaderLink = ({ children, className, exact, href, ...props }) => {
-  const classes = classnames(['header-link', className]);
+export const HeaderLink = ({ children, className, href, link, ...props }) => {
```
Comment (mabhub) [inline src/components/HeaderLink.js:5]: Also, I would invert return conditions:

```jsx
export const HeaderLink = ({ children, className, href, link, ...props }) => {
  const attributes = {
    className: classnames(['header-link', className]),
    ...props,
  };

  if (!href) {
    return <span {...attributes}>{children}</span>;
  }

  if (!link || href.startsWith('http')) {
    return <a href={href} {...attributes}>{children}</a>;
  }

  const {
    component: Link,
    linkProps: { hrefAttribute = 'href', ...linkProps },
  } = link;

  const attributesLink = {
    ...attributes,
    [hrefAttribute]: href,
    ...linkProps,
  };

  return <Link {...attributesLink}>{children}</Link>;
};
```

**Comment by dtrucs** [comment]
As seen IRL with @mabhub, we could refacto the whole component by disabling the test checking if the `href` starts with `http` string (test not really useful).
```jsx
export const HeaderLink = ({ className, href, link, ...props }) => {
  const {
    component: Component,
    linkProps: { hrefAttribute, ...linkProps },
  } = link || {
    component: href ? 'a' : 'span',
    linkProps: { hrefAttribute: 'href' },
  };

  const hrefProp = href && {
    [hrefAttribute]: href,
  };

  return (
    <Component
      className={classnames(['header-link', className])}
      {...hrefProp}
      {...linkProps}
      {...props}
    />
  );
};
```

**Review by jrmi — APPROVED** [review]

**Review by jrmi — APPROVED** [review]

**Pull request merged**